ambassador/ambassador_diag/diagd.py

```diff
@@ -965,68 +1030,92 @@ def chime(self):
+        env_good = True
+        chime_failures = {}
+        env_status = SystemStatus()
+
+        error_count = 0
+        tls_count = 0
+        mapping_count = 0
+
         if not ir:
             ir = app.ir
 
-        env_good = True
-        failures = {}
-
         if not ir:
-            failures['no config loaded'] = True
+            chime_failures['no config loaded'] = True
             env_good = False
         else:
             if not ir.aconf:
-                failures['completely empty config'] = True
+                chime_failures['completely empty config'] = True
                 env_good = False
             else:
                 for err_key, err_list in ir.aconf.errors.items():
                     if err_key == "-global-":
                         err_key = ""
 
                     for err in err_list:
+                        error_count += 1
                         err_text = err['error']
 
                         self.app.logger.info(f'error {err_key} {err_text}')
 
                         if err_text.find('CRD') >= 0:
                             if err_text.find('core') >= 0:
-                                failures['core CRDs'] = True
+                                chime_failures['core CRDs'] = True
+                                env_status.failure("CRDs", "Core CRD type definitions are missing")
                             else:
-                                failures['other CRDs'] = True
+                                chime_failures['other CRDs'] = True
+                                env_status.failure("CRDs", "Resolver CRD type definitions are missing")
 
                             env_good = False
                         elif err_text.find('TLS') >= 0:
-                            failures['TLS errors'] = True
-                            env_good = False
+                            chime_failures['TLS errors'] = True
+                            env_status.failure('TLS', err_text)
 
-            some_tls = False
+                            env_good = False
 
             for context in ir.tls_contexts:
                 if context:
-                    some_tls = True
+                    tls_count += 1
                     break
 
-            if not some_tls:
-                failures['no TLS contexts'] = True
-                env_good = False
+            for group in ir.groups.values():
+                for mapping in group.mappings:
+                    pfx = mapping.get('prefix', None)
+                    name = mapping.get('name', None)
 
-            some_mappings = False
+                    if pfx:
+                        if not pfx.startswith('/ambassador/v0') or not name.startswith('internal_'):
+                            mapping_count += 1
 
-            for group in ir.groups.values():
-                if group and (group.location != '--internal--'):
-                    some_mappings = True
-                    break
+        if error_count:
```

Comment:
Do we want this instead?

```suggestion
        if error_count > 0:
```